### ISSUE TITLE
annot.next throws AttributeError

### ISSUE THREAD

**Issue by Luux** [pymupdf/PyMuPDF#2969]
### Description of the bug

Using the `next` property of an annotation throws the following error:
```
    @property
    def next(self):
        """Next annotation."""
        CheckParent(self)
        this_annot = self.this
        assert isinstance(this_annot, mupdf.PdfAnnot)
        assert this_annot.m_internal
        type_ = mupdf.pdf_annot_type(this_annot)
        if type_ != mupdf.PDF_ANNOT_WIDGET:
            annot = mupdf.pdf_next_annot(this_annot)
        else:
            annot = mupdf.pdf_next_widget(this_annot)
    
        val = Annot(annot) if annot.m_internal else None
        if not val:
            return None
        val.thisown = True
>       assert val.parent.this.m_internal_value() == self.parent.this.m_internal_value()
E       AttributeError: 'Annot' object has no attribute 'parent'
```

This only occurs with the latest release candidate. It works with the previous version and also works with `fitz_old`.

### How to reproduce the bug

[annotated.pdf](https://github.com/pymupdf/PyMuPDF/files/13820505/annotated.pdf)

```
import fitz
doc = fitz.open("annotated.pdf")
page = doc[0]
first_annot = list(page.annots())[0]
first_annot.next
```

### PyMuPDF version

1.23.9rc1

### Operating system

Linux

### Python version

3.12

**Comment by julian-smith-artifex-com** [comment]
Thanks for this. I have a fix in my tree, hopefully it'll pass internal review soon and be in the next release.

**Comment by julian-smith-artifex-com** [comment]
This is fixed in release candidate 1.23.9rc2; see: #2996

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.9.
